kotlin-sdk-core/src/commonMain/kotlin/io/modelcontextprotocol/kotlin/sdk/types.kt

```diff
@@ -975,19 +975,45 @@ public data class GetPromptRequest(
 }
 
 /**
- * Represents the content of a prompt message.
+ * The server's response content indicating the type of content.
  */
-@Serializable(with = PromptMessageContentPolymorphicSerializer::class)
-public sealed interface PromptMessageContent {
+public sealed interface ContentWithType {
     public val type: String
 }
 
 /**
- * Represents prompt message content that is either text, image or audio.
+ * Represents the content of a prompt message. Deprecated.
  */
-@Serializable(with = PromptMessageContentMultimodalPolymorphicSerializer::class)
+@Deprecated("For backwards compatibility; use ContentBlock instead", ReplaceWith("ContentBlock"))
+public sealed interface PromptMessageContent : ContentWithType
+
+/**
+ * Represents multimodal content of a prompt message. Deprecated.
+ */
+@Deprecated(
+    "For backwards compatibility; use CreateMessageResultContent or SamplingMessageContent instead",
+    ReplaceWith("CreateMessageResultContent"),
+)
 public sealed interface PromptMessageContentMultimodal : PromptMessageContent
 
+/**
+ * Represents the types of a ContentBlock
+ */
+@Serializable(with = ContentBlockPolymorphicSerializer::class)
+public sealed interface ContentBlock : PromptMessageContent
+
+/**
+ * Represents content for the CreateMessageResult
+ */
+@Serializable(with = CreateMessageResultContentMultimodalPolymorphicSerializer::class)
+public sealed interface CreateMessageResultContent : ContentWithType
+
+/**
+ * Represents content for the SamplingMessage
+ */
+@Serializable(with = SamplingMessageContentMultimodalPolymorphicSerializer::class)
+public sealed interface SamplingMessageContent : ContentWithType
```

Comment:
It’s not clear to me why these interfaces were separated and added, what’s the purpose?
Also, since custom serializers were defined for them, wouldn’t deserializers be required as well?